internals/secrethub/init.go

```diff
@@ -105,12 +121,98 @@ func (cmd *InitCommand) Run() error {
 	case InitModeSignup:
 		signupCommand := SignUpCommand{
 			io:              cmd.io,
-			newClient:       cmd.newClient,
+			newClient:       cmd.newUnauthenticatedClient,
 			credentialStore: cmd.credentialStore,
 			progressPrinter: cmd.progressPrinter,
 			force:           cmd.force,
 		}
 		return signupCommand.Run()
+	case InitModeSetupCode:
+		setupCode := cmd.setupCode
+		if setupCode == "" {
+			var err error
+			setupCode, err = ui.AskAndValidate(cmd.io, "What is your setup code?\n", 3, api.ValidateSetupCode)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintf(
+			cmd.io.Output(),
+			"An account credential will be generated and stored at %s. "+
+				"Losing this credential means you lose the ability to decrypt your secrets. "+
+				"So keep it safe.\n",
+			credentialPath,
+		)
+
+		// Only prompt for a passphrase when the user hasn't used --force.
+		// Otherwise, we assume the passphrase was intentionally not
+		// configured to output a plaintext credential.
+		var passphrase string
+		if !cmd.credentialStore.IsPassphraseSet() && !cmd.force {
+			var err error
+			passphrase, err = ui.AskPassphrase(cmd.io, "Please enter a passphrase to protect your local credential (leave empty for no passphrase): ", "Enter the same passphrase again: ", 3)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Can we de-duplicate this?

Reply:
I couldn't come up with a meaningful method for deduplicating this. Maybe we could factor out all the prompting and credential writing the same way we moved the environment related stuff out of the run command.

Reply:
Hmm, okay, de-duplicating the control logic is maybe not straightforward nor would it improve the code quality. But then we should at least de-duplicate the strings used. Either by making them constants or - I think preferably - moving this to a helper function that is called here and in the backup code initialization. Something like:
```golang
func askCredentialPassphrase(io ui.IO) (error, string){
    return ui.AskPassphrase(io, "Please enter a passphrase to protect your local credential (leave empty for no passphrase): ", "Enter the same passphrase again: ", 3)
}
```

Reply:
Good idea! I implemented it.